tests/test_generics.py

```diff
@@ -2446,3 +2454,27 @@ class VariadicToolbox(BaseModel, Generic[ComponentVar, Unpack[NumberOfComponents
     with pytest.raises(TypeError):
         # Should raise error because integer 5 does not meet the bound requirements of ComponentVar
         MyToolboxClass(manual_toolset=(sa, sb, 5))
+
+
+@pytest.mark.skipif(
```

Comment:
Does it work with typing extensions?

Reply:
I added support for recursive namedtuple in the latest commit on this branch, and as part of that, added a test that used `typing_extensions.NamedTuple` to get generics support in older versions of python. It's working for all older versions, so I'm pretty sure it's good.

(The change I made was fairly small, essentially just copied the contextmanager that was used in the typeddict_schema to handle recursion)